### PR TITLE
User settings template for FIPS v5 ready

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Template for building FIPS v5 ready with `user_settings.h`.
Fixes ZD 13538

# Testing

```sh
./fips-check.sh linuxv5-dev keep
cd XXX-fips-test
cp ../examples/configs/user_settings_fipsv5.h user_settings.h
./configure --enable-fips=v5-dev --enable-usersettings
make
./fips-hash.sh
make
make check
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [X] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: examples/configs/user_settings_fipsv5.h [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#define WOLFSSL_BASE64_ENCODE
+#define WOLFSSL_USE_ALIGN
+#define WOLFSSL_PUBLIC_MP
+#define HAVE_ENCRYPT_THEN_MAC
+#define HAVE_TLS_EXTENSIONS
+#define HAVE_SUPPORTED_CURVES
+#define HAVE_EXTENDED_MASTER
+//#define HAVE_SERVER_RENEGOTIATION_INFO
+
+#undef  WOLFSSL_WOLFSSH
```
Comment (douzzer) [inline examples/configs/user_settings_fipsv5.h:74]: did you mean to keep this `#undef`?

**Review by douzzer — CHANGES_REQUESTED** [review]
LGTM except for the one nit.

File: examples/configs/user_settings_fipsv5.h (hunk 1 above)
Comment (dgarske) [reply examples/configs/user_settings_fipsv5.h]: Yes, for some reason the configure puts it in the cflags and was getting a re-def. Only this one.

File: examples/configs/user_settings_fipsv5.h [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+/* TLS Resumption */
+#define HAVE_SECURE_RENEGOTIATION
+#define HAVE_SERVER_RENEGOTIATION_INFO
+#define HAVE_SESSION_TICKET
+
+/* Other Features */
+#define WOLFSSL_USE_ALIGN
+#define WOLFSSL_BASE64_ENCODE
+//#define HAVE_CRL
```
Comment (SparkiDev) [inline examples/configs/user_settings_fipsv5.h:80]: Comment why this disabled or why a customer would enable.

File: examples/configs/user_settings_fipsv5.h [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+#define WOLFSSL_ALWAYS_VERIFY_CB
+#define WOLFSSL_DH_EXTRA /* DHE ASN.1 key import/export support */
+
+#ifndef WOLFSSL_WOLFSSH /* ifndef check required due to configure already generating flag */
+#define WOLFSSL_WOLFSSH
+#endif
+
+/* Compatibility Layer */
+#define OPENSSL_EXTRA
+#if 0
```
Comment (SparkiDev) [inline examples/configs/user_settings_fipsv5.h:92]: Comment on when to enable.

File: examples/configs/user_settings_fipsv5.h [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+#define HAVE_ECC521
+#define WOLFSSL_ECDSA_SET_K
+
+/* RSA */
+#undef  NO_RSA
+#define WC_RSA_PSS
+#define WOLFSSL_KEY_GEN
+#define WC_RSA_NO_PADDING
+
+/* AES */
```
Comment (SparkiDev) [inline examples/configs/user_settings_fipsv5.h:126]: #undef NO_AES

File: examples/configs/user_settings_fipsv5.h [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+#define NO_RC4
+#define NO_MD4
+#define NO_MD5
+#define NO_DES3
+#define NO_DSA
+#define NO_RABBIT
+#define NO_HC128
+#define WOLFSSL_NO_SHAKE256
+
+#if 0
```
Comment (SparkiDev) [inline examples/configs/user_settings_fipsv5.h:167]: Comment on when to enable.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by SparkiDev — APPROVED** [review]

**Review by SparkiDev — APPROVED** [review]

**Review by douzzer — APPROVED** [review]
LGTM.  As noted, `#define HAVE_CRL` is omitted from user_settings_fipsv5.h, so `scripts/crl-revoked.test` fails, but this was deemed OK.  All other tests succeed.

**Pull request merged**